### DISCUSSION TITLE
Welcome to x32Live-CleanUp Discussions!

### DISCUSSION THREAD

**Discussion by Topslakr** [Topslakr/x32Live-CleanUp#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jcavaz** [comment]
Hi there,
Thanks for awesome script!
I've been writing a similar script as a batch file.  It copies the sdcard wav files, splits them, and then joins them into 32 individual files. The cleanup feature you have was not easy to translate into a batch file. I'd like to learn how to speed it up.  It takes over 1 hour to run for a one hour recording, or 8 full wave files.
I'm having an issue using the file output in Cakewalk. Although, it works fine in Cubase.  It probably has something to do with the codec in FFMPEG.  I've attached the text of the file I wrote.  It's a batch file that would need the extension change to .bat.  What other codec would you suggest?

[WavSplitter.txt](https://github.com/Topslakr/x32Live-CleanUp/files/6218951/WavSplitter.txt)

**Reply by Topslakr** [reply]
This is cool! Glad I was able to offer someone at least a bit of help...

Have you tried running this NOT on an SD card? I would be willing to bet your big slowdown is down to the write speed of the SD card. It needs to write a lot of data and it's in multiple streams. This is very very slow on SD cards. This is, in fact, why the X32 creates that one big multi-channel file.

Can you try running it on a HDD/SSD and see how long it takes?

**Reply by jcavaz** [reply]
Thanks! It's set to run on one of my SSD drives.  The delay is probably due to running ffmpeg one file at a time. It takes 12 seconds per track.  12seconds x 32tracks x 9 Files takes about 1 hour.  I'm going to run a seperate script to handle half of the original files and launch it after the sd card files are copied.  I'll have to pause the script before running Sox and make sure all files are ready to be joined in sequence.  I got Cakewalk to work with a different codec.  I used the adpcm_ms codec.  It's not the full raw recording but for right now it will work for me.  Here's the file I used yesterday.

[3-28-2021 WavSplitter.txt](https://github.com/Topslakr/x32Live-CleanUp/files/6224786/3-28-2021.WavSplitter.txt)

**Reply by Topslakr** [reply]
I did some testing today with ffmpeg in my script today, and no matter what I set for threads, it only uses one CPU. You, though, are running the ffmpeg command 32 times. You might have better luck running the script on each 000000X.wav file, instead of running them in series one after the other.

I also wonder if you would see higher speeds if you ran one large ffmpeg command, like in my script. In my testing one big command was faster than running it 32 times in series.

**Reply by jcavaz** [reply]
I was able to get it to run faster by splitting the script up.  It saved about a half hour running two scripts at the same time.  I tried cutting the script into 3 files but it didn't save any more time.

**Comment by kenironside** [comment]
Hi, two questions:
One: I think I'm reading the info incorrectly but how do I apply your Split script to multiple multichannel wav files from the same recording? I have been using Reaper to unpack the wav files and explode them to 32 channel wav files but would like to use a simpler method, which is what I think you have created.
 Two: Today I had a longer recording session than usual and the multichannel wav's were named 0000001-9 and then 000000A and 000000B. Reaper decided that the A & B files come before the 1-9 files and the start of my recording was 3/4 or the way through the church service! I had to find the start of the service and move it to the beginning of the recording! Does your software work that way?
Thanks for considering basic questions and thank you for making this tool available. Regards, Ken

**Reply by Topslakr** [reply]
Interesting Questions. Thanks for asking! I too use it to split apart the audio after church services, though I found Audacity faster to use than Reaper for this task. Still too slow and cumbersome though for me, which is why I wrote this script.

One:
So, the script is pretty simple. It'll only process one file at a time, not a batch of files. I'd like to add that to the script, but I'm not there yet. So you'd run the 'Split' script for each file you have, one after another.

You could do "Split 0000001.wav && Split 0000002.wav", etc., to have them all just run one after the other.

Two:
So, since the script I wrote is, again, very simple, it's not going to do anything quite as fancy as I think you were hoping for. If you hand it a file that ends in 01, or 0A, it's going to use those last two characters to prefix the output.

So, File 0000001.wav will be split into 01-01.wav, 01-02.wav ---> 01-32.wav
And, file 000000A.wav will be split into 0A-01.wav, 0A-02.wav ---> 0A-32.wav

That being said, personally I would rename the lettered files with digits instead. So, after 0000009.wav, I would rename 000000A.wav to 0000010.wav, etc., so that Reaper, or whatever DAW you use, will understand the order.

I hope that helps. Good luck!

**Reply by kenironside** [reply]
Thanks! Perfectly clear. Appreciate your taking time to respond.
